src/components/nabla/Pools/Backstop/WithdrawLiquidity/useWithdrawLiquidity.ts

```diff
@@ -2,91 +2,159 @@ import { yupResolver } from '@hookform/resolvers/yup';
+          return 'The input amount is too large.';
+      }
+      return 'Cannot determine value of shares';
+  }
+}
 
 const storageSet = debounce(storageService.set, 1000);
-export const useWithdrawLiquidity = (pool: BackstopPool) => {
-  const { id: poolAddress, token, router } = pool;
-  const tokenAddress = token.id;
-  const swapPools = router?.swapPools;
-  const { indexerUrl } = useGetAppDataByTenant('nabla').data || {};
+
+export const useWithdrawLiquidity = (nabla: NablaInstance) => {
   const queryClient = useQueryClient();
   const toggle = useModalToggle();
-  const tokenModal = useBoolean();
 
-  const balanceQuery = useContractBalance({ contractAddress: tokenAddress, decimals: FixedU128Decimals });
-  const depositQuery = useContractBalance({ contractAddress: poolAddress, decimals: FixedU128Decimals });
-  const { refetch: balanceRefetch } = balanceQuery;
-  const { refetch: depositRefetch } = depositQuery;
+  const backstopPoolAddress = nabla.backstopPool.id;
+
+  const swapPools = nabla.swapPools;
 
   const form = useForm<WithdrawLiquidityValues>({
     resolver: yupResolver(schema),
-    defaultValues: {},
+    defaultValues: getInitialValues(),
   });
-  const { reset, getValues, handleSubmit } = form;
-  const amount =
-    Number(
-      useWatch({
-        control: form.control,
-        name: 'amount',
-        defaultValue: 0,
-      }),
-    ) || 0;
 
   const address = useWatch({
     control: form.control,
     name: 'address',
   });
 
-  const pools = useMemo(
-    () =>
-      [
-        {
-          id: '',
-          token: {
-            ...token,
-            id: '',
-          },
-        } as SwapPool,
-      ].concat(swapPools || []),
-    [swapPools, token],
+  const selectedPool = useMemo<NablaInstanceSwapPool | undefined>(
+    () => swapPools.find((t) => t.id === address)!,
+    [address, swapPools],
   );
-  const selectedPool = useMemo(() => pools.find((t) => t.id === address) || pools[0], [address, pools]);
 
+  const tokenToReceive = selectedPool?.token ?? nabla.backstopPool.token;
+
+  const balanceQuery = useErc20ContractBalance(erc20WrapperAbi, {
+    contractAddress: tokenToReceive.id,
+    decimals: tokenToReceive.decimals,
+  });
+
+  const depositQuery = useErc20ContractBalance(backstopPoolAbi, {
+    contractAddress: backstopPoolAddress,
+    decimals: nabla.backstopPool.lpTokenDecimals,
+  });
+
+  const { reset, getValues } = form;
+  const amountString = useWatch({
+    control: form.control,
+    name: 'amount',
+    defaultValue: '0',
+  });
+
+  const { refetch: balanceRefetch } = balanceQuery;
+  const { refetch: depositRefetch } = depositQuery;
+  const { refetchQueries } = queryClient;
   const onWithdrawSuccess = useCallback(() => {
     reset();
     balanceRefetch();
     depositRefetch();
-    queryClient.refetchQueries([cacheKeys.backstopPools, indexerUrl]);
-  }, [balanceRefetch, depositRefetch, indexerUrl, queryClient, reset]);
+    setTimeout(() => {
```

Comment:
Could we apply the same fix about not using setTimeout here as well? It would help us maintain consistency and avoid potential issues across the application.

Reply:
Here we query the smart contract and don't need to delay as the smart contract will already have its updated value at this point (unlike the indexer).